### PR TITLE
added cleanup command

### PR DESCRIPTION
**Pull request by professor91**

### CODE REVIEW

File: server/src/testing/testingRoutes.ts [hunk 1]
```diff
@@ -128,5 +128,59 @@ export default function testingHandler(server: FastifyInstance, options, done) {
     console.log(res)
     reply.code(200).send(res)
   })
+  server.get('/genData', async (req, reply) => {
```
Comment (rollie42) [inline server/src/testing/testingRoutes.ts:131]: Can we guard this so it's only accessible in non-dev env?

File: server/src/testing/testingRoutes.ts [hunk 2]
```diff
@@ -128,5 +128,59 @@ export default function testingHandler(server: FastifyInstance, options, done) {
...
+        teamLead: false,
+      })
+
+      console.log('database seed was successful');
+    } catch (error) {
+      console.log(error, 'database seed failed');
+    }
+  })
+  server.get('/testcleanup', async (req, reply) => {
+    const targetTeam = await firebaseApi.getTeam('test_role')
```
Comment (rollie42) [inline server/src/testing/testingRoutes.ts:169]: This doesn't clean up the 2nd team

File: server/src/hwApi.ts [hunk 3]
```diff
@@ -321,4 +321,38 @@ removed from your team and need to be re-added if you wish to participate in fut
...
+    const users = await firebaseApi.getUsers()
+    const teamMembers = users.filter(
+      (u) => u.team === targetTeam.id
+    )
+
+    for (const user of teamMembers) {
+      await this.removeUserFromTeam(user)
+
+      if (user.teamLead) {
+        await this.setTeamLead(user, false)
```
Comment (rollie42) [inline server/src/hwApi.ts]: removeUserFromTeam should be changed to do this

File: server/src/admin/commands/admin-commands/CleanUpTeam.ts [hunk 4]
```diff
@@ -0,0 +1,29 @@
+import { RegisteredUserAppCommand } from "../DiscordAppCommand.js";
```
Comment (rollie42) [inline server/src/admin/commands/admin-commands/CleanUpTeam.ts]: Every command filename is "DoSomethingCommand.ts" - should rename this file appropriately

File: server/src/hwApi.ts [hunk 5]
```diff
@@ -321,4 +321,38 @@ removed from your team and need to be re-added if you wish to participate in fut
       message: 'Github id updated',
     }
   }
+  async cleanupTeam(teamRole: string) {
```
Comment (rollie42) [inline server/src/hwApi.ts]: If you look  at the other functions in this API, they are all like:

async addUserToTeam(user: UserT, team: TeamT, silent = false) {
async removeUserFromTeam(user: UserT) {

etc. We don't use identifiers here, so instead have it accept `(team: TeamT)`

File: server/src/testing/testingRoutes.ts (hunk 1 above)
Comment (rollie42) [reply server/src/testing/testingRoutes.ts:131]: Actually, we should block this entire handler in prod - so changing it in server.ts line 65 to only register in dev environment

File: server/src/discord/discordApi.ts [hunk 6]
```diff
@@ -189,6 +189,17 @@ ${await teamsSection()}${await usersSection()}
       console.log(`Failed to update LFG post: ${e}`)
     }
   }
+  
+  async resetTeamDefaultChannel(targetTeam: TeamT) {
```
Comment (rollie42) [inline server/src/discord/discordApi.ts]: Similar to how hwApi works off the types defined in the app, *this* file only works with discord "types".  Imagine if this were shared with other users as an API to access discord APIs. If you have a tic-tac-toe game, there's no concept of a 'TeamT'. Also this API shouldn't be calling things inside the firebaseAPI (I did break this rule for the LFG/LFM stuff, which perhaps should be changed as well, but it's fine to leave that part alone for your change)

I think if this file is exposing discord functionality, it's fine to have a 'recreateChannel' function that gets the category, etc, and then just returns the new channel ID, which is saved at the call site for this function, presumably in hwApi.

File: server/src/testing/testingRoutes.ts (hunk 2 above)
Comment (professor91) [reply server/src/testing/testingRoutes.ts:169]: didn't get it...the script is supposed to mimic the cleanupcommand which basically takes a team and resets it, it does not handle multiple teams input

File: server/src/discord/discordApi.ts (hunk 6 above)
Comment (professor91) [reply server/src/discord/discordApi.ts]: agreed I will update the params

about second point u mentioned `file`...iiuc `recreateChannel` should be a static method of discordApi, right?

File: server/src/testing/testingRoutes.ts (hunk 2 above)
Comment (rollie42) [reply server/src/testing/testingRoutes.ts:169]: Ah I see what this is intending - okay to leave as is

File: server/src/discord/discordApi.ts (hunk 6 above)
Comment (rollie42) [reply server/src/discord/discordApi.ts]: There aren't any static methods, it should just be a method in discordApi like any other, that takes in a channel id, and returns the newly created channel id

**Review by rollie42 — APPROVED** [review]

**Pull request merged**